### DISCUSSION TITLE
Uninstall didn't clear everything and now I cannot reinstall :-/ frustrated and need any help pls :)

### DISCUSSION THREAD

**Discussion by mannp** [longhorn/longhorn#4986] [General]
`v1.24.6+k3s1 / lh helm chart 1.3.2`

Uninstalled longhorn using helm and am left with crds and some volumes I am unable to delete or force delete.

Trying to reinstall I get thee following error and any help would be greatly appreciated after an afternoon trying to sort it :-/

**Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install:** 

`CustomResourceDefinition "engines.longhorn.io" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "longhorn"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "longhorn-system"`

Also, why am I getting annotation validation errors from the standard helm install? Is it borked? Got these errors with the following command;

`helm install longhorn longhorn/longhorn --namespace longhorn-system --create-namespace --version 1.3.2`

Tried this from the manual troubleshooting with no joy;
```
Problems with CRDs
If your CRD instances or the CRDs themselves can’t be deleted for whatever reason, run the commands below to clean up. Caution: this will wipe all Longhorn state!

# Delete CRD finalizers, instances and definitions
for crd in $(kubectl get crd -o jsonpath={.items[*].metadata.name} | tr ' ' '\n' | grep longhorn.rancher.io); do
  kubectl -n ${NAMESPACE} get $crd -o yaml | sed "s/\- longhorn.rancher.io//g" | kubectl apply -f -
  kubectl -n ${NAMESPACE} delete $crd --all
  kubectl delete crd/$crd
done
```
Also tried -> kubectl delete crd/engines.longhorn.io --force & kubectl delete crd/volumes.longhorn.io with no change

The following volumes cannot be deleted and appear even when the namespace is deleted;

```
volume.longhorn.io/pvc-01641e66-cf15-4f7b-afe7-e3190d390666   attached   healthy                  1073741824   inst-d8kpt-k3s-workers   8d
volume.longhorn.io/pvc-138b9b75-8f1c-4962-a94b-616a911cfab3   detached   unknown                  3221225472                            11d
volume.longhorn.io/pvc-63fc826d-19c5-4b9f-88bf-5df47c7b0476   attached   healthy                  419430400    inst-iezhh-k3s-workers   8d
volume.longhorn.io/pvc-9b0a737f-a591-497a-a512-c24a132e494b   detached   unknown                  134217728                             10d
volume.longhorn.io/pvc-9f0c6d0b-8547-4608-901a-81b557f609bf   attached   healthy                  134217728    inst-iezhh-k3s-workers   26d
volume.longhorn.io/pvc-b7df8092-f413-49d7-a9ce-8a9d2c0a18bd   detached   unknown                  3221225472                            11d
volume.longhorn.io/pvc-cfe4fdc9-733e-4925-9259-f4560ee8495e   detached   unknown                  3221225472                            11d
volume.longhorn.io/pvc-e8e925b6-8604-4b24-97dc-3c9b0f3adec5   attached   healthy                  1073741824   inst-d8kpt-k3s-workers   8d
volume.longhorn.io/pvc-ee503c71-30a4-4ff9-9e2a-980375bcf02c   detached   unknown                  134217728                             10d
volume.longhorn.io/pvc-f31ce83c-2d50-4c09-8b96-1375f3bbd85d   attached   healthy                  134217728    inst-cmrfo-k3s-servers   10d
volume.longhorn.io/pvc-f4749cf9-316c-4ab5-bfd7-2f24c8e59723   attached   healthy                  134217728    inst-cmrfo-k3s-servers   10d

```

**Comment by mannp** [comment]
Resolved by editing the finalisers for all the stuck resources. Tedious, but worked.

**Comment by mantissahz** [comment]
@mannp Could you kindly provide the install Longhorn and create volumes steps and uninstall steps?
I could not reproduce it in my local environment `v1.24.6+k3s1 / lh helm chart 1.3.2 with ubuntu 2204 VM` .
